static/js/tools/shared/metadata/metadata_fetcher.ts

```diff
@@ -448,32 +464,50 @@ export async function fetchMetadata(
       const key = matchedSeries?.seriesKey ?? {};
       const unit = key.unit || facetInfo.unit;
 
+      const sourceName = provenanceData?.source?.[0]?.name;
+      const provenanceName =
+        provenanceData?.isPartOf?.[0]?.name ||
+        provenanceData?.name?.[0]?.value ||
+        importName;
```

Comment:
(for later): rather than duplicate this logic across the file, could we move this logic upstream into flask? please add a TODO for now.